packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts

```diff
@@ -1,41 +1,368 @@
+		} else if (binding.type === "send_email") {
+			bindings.send_email ??= [];
+			bindings.send_email.push({ ...binding, name: name });
+		} else if (binding.type === "wasm_module") {
+			bindings.wasm_modules ??= {};
+			bindings.wasm_modules[name] = await getBinaryFileContents(binding.source);
+		} else if (binding.type === "text_blob") {
+			bindings.text_blobs ??= {};
+			bindings.text_blobs[name] = binding.source.path as string;
+		} else if (binding.type === "data_blob") {
+			bindings.data_blobs ??= {};
+			bindings.data_blobs[name] = await getBinaryFileContents(binding.source);
+		} else if (binding.type === "browser") {
+			bindings.browser = { binding: name };
+		} else if (binding.type === "ai") {
+			bindings.ai = { binding: name };
+		} else if (binding.type === "version_metadata") {
+			bindings.version_metadata = { binding: name };
+		} else if (binding.type === "durable_object_namespace") {
+			bindings.durable_objects ??= { bindings: [] };
+			bindings.durable_objects.bindings.push({ ...binding, name: name });
+		} else if (binding.type === "queue") {
+			bindings.queues ??= [];
+			bindings.queues.push({ ...binding, binding: name });
+		} else if (binding.type === "r2_bucket") {
+			bindings.r2_buckets ??= [];
+			bindings.r2_buckets.push({ ...binding, binding: name });
+		} else if (binding.type === "d1") {
+			bindings.d1_databases ??= [];
+			bindings.d1_databases.push({ ...binding, binding: name });
+		} else if (binding.type === "vectorize") {
+			bindings.vectorize ??= [];
+			bindings.vectorize.push({ ...binding, binding: name });
+		} else if (binding.type === "constellation") {
+			bindings.constellation ??= [];
+			bindings.constellation.push({ ...binding, binding: name });
+		} else if (binding.type === "hyperdrive") {
+			bindings.hyperdrive ??= [];
+			bindings.hyperdrive.push({ ...binding, binding: name });
+		} else if (binding.type === "service") {
+			bindings.services ??= [];
+			bindings.services.push({ ...binding, binding: name });
+		} else if (binding.type === "fetcher") {
+			fetchers[name] = binding.fetcher;
+		} else if (binding.type === "analytics_engine") {
+			bindings.analytics_engine_datasets ??= [];
+			bindings.analytics_engine_datasets.push({ ...binding, binding: name });
+		} else if (binding.type === "dispatch_namespace") {
+			bindings.dispatch_namespaces ??= [];
+			bindings.dispatch_namespaces.push({ ...binding, binding: name });
+		} else if (binding.type === "mtls_certificate") {
+			bindings.mtls_certificates ??= [];
+			bindings.mtls_certificates.push({ ...binding, binding: name });
+		} else if (binding.type === "logfwdr") {
+			bindings.logfwdr ??= { bindings: [] };
+			bindings.logfwdr.bindings.push({ ...binding, name: name });
+		} else if (binding.type.startsWith("unsafe-")) {
+			bindings.unsafe ??= {
+				bindings: [],
+				metadata: undefined,
+				capnp: undefined,
+			};
+			bindings.unsafe.bindings?.push({ ...binding, name: name });
```

Comment:
This is missing the `type` property transformation from `unsafe-x` to just `x`

Reply:
https://github.com/cloudflare/workers-sdk/pull/4450/commits/d7c2bd7c7755036a74be539ddefabe42516c2384